DS13/code/modules/overmap/admin.dm

```diff
@@ -0,0 +1,67 @@
+
+/obj/structure/overmap_component/helm/spawner //A helm console which when clicked by an admin, will spawn a new overmap ship (unless it's already linked to one!) This is used for the preset adminbus ships.
+	var/ship_type = null //Set this to !null and it'll spawn a ship when clicked if one doesn't exist.
+
+/obj/structure/overmap_component/helm/spawner/dderidex
+	ship_type = /obj/structure/overmap/dderidex
+	redalert_sound = 'DS13/sound/effects/redalert-rom.ogg'
+
+/obj/structure/overmap_component/helm/spawner/attack_hand(mob/user)
+	if(!linked && ship_type) //If we have something to make, make it.
+		if(check_rights_for(user.client, R_ADMIN))
+			var/question = alert("This area does not have a linked ship, would you like to create one?",name,"Yes","No")
+			if(question == "No" || !question)
+				return
+			var/obj/effect/landmark/overmap_event_spawn/S = pick(GLOB.overmap_event_spawns)
+			linked = new ship_type(get_turf(S))
```

Comment:
What is `S` in this context?

Reply:
Temporary var for choosing a random event spawn, we don't need relevent names for temp vars for the most part.

Reply:
I don't see the connection between `S` and "random event spawn".